services/rollout-service/pkg/versions/versions.go

```diff
@@ -132,6 +144,16 @@ outer:
 					}
 				}
 			}
+			for k := range versions {
+				if seen[k] == 0 {
+					processor.ProcessKuberpultEvent(ctx, KuberpultEvent{
```

Comment:
What are really achieving here? Calling processor.ProcessKuberpultEvent(version=0) for all apps previously processed?
Maybe add a comment to explain why that is necessary. Or rename `seen`, it seems to be used only for the 0-version. Maybe `processZeroVersion`.

Reply:
Ah, right, this is for taking care of removed applications. If we don't send a zero value, the rollout service would consider  them deployed for-ever and would always reconsider them for notification. I will write a comment on that.

Reply:
Maybe also mention this in the committe message

Reply:
Yes, I added this to the message and added a comment.